logging/tests/unit/handlers/test__helpers.py

```diff
@@ -0,0 +1,122 @@
+        from google.cloud.logging.handlers._helpers import get_trace_id
+
+        with self.app.test_request_context(
+                path='/',
+                headers={}):
+            trace_id = get_trace_id_from_flask()
+            trace_id_returned = get_trace_id()
+
+        self.assertEqual(trace_id, None)
+        self.assertEqual(trace_id_returned, None)
+
+    def test_trace_id_valid_context_header(self):
+        from google.cloud.logging.handlers._helpers import get_trace_id_from_flask
+        from google.cloud.logging.handlers._helpers import get_trace_id
+
+        flask_trace_header = 'X_CLOUD_TRACE_CONTEXT'
+        flask_trace_id = 'testtraceidflask/testspanid'
+
+        with self.app.test_request_context(
+                path='/',
+                headers={flask_trace_header:flask_trace_id}):
+            trace_id = get_trace_id_from_flask()
+            trace_id_returned = get_trace_id()
+
+        expected_trace_id = 'testtraceidflask'
+        self.assertEqual(trace_id, expected_trace_id)
+        self.assertEqual(trace_id, trace_id_returned)
+
+
+class TestDjangoTrace(unittest.TestCase):
+
+    def setUp(self):
+        from django.conf import settings
+        from django.test.utils import setup_test_environment
+
+        if not settings.configured:
+            settings.configure()
+        setup_test_environment()
+
+    def tearDown(self):
+        from django.test.utils import teardown_test_environment
+
+        teardown_test_environment()
+
+    def test_trace_id_no_context_header(self):
+        from django.test import RequestFactory
+        from google.cloud.logging.handlers._helpers import get_trace_id_from_django
+        from google.cloud.logging.handlers._helpers import get_trace_id
+        from google.cloud.logging.handlers.middleware.request import RequestMiddleware
+        from google.cloud.logging.handlers.middleware.request import _thread_locals
+
+        request = RequestFactory().get('/')
+
+        middleware = RequestMiddleware()
+        middleware.process_request(request)
+        trace_id = get_trace_id_from_django()
+        trace_id_returned = get_trace_id()
+
+        self.assertEqual(trace_id, None)
+        self.assertEqual(trace_id_returned, None)
+
+        _thread_locals.__dict__.clear()
+
+    def test_trace_id_valid_context_header(self):
+        from django.test import RequestFactory
+        from google.cloud.logging.handlers._helpers import get_trace_id_from_django
+        from google.cloud.logging.handlers._helpers import get_trace_id
+        from google.cloud.logging.handlers.middleware.request import RequestMiddleware
+        from google.cloud.logging.handlers.middleware.request import _thread_locals
+
+        django_trace_header = 'HTTP_X_CLOUD_TRACE_CONTEXT'
+        django_trace_id = 'testtraceiddjango/testspanid'
+
+        request = RequestFactory().get(
+            '/',
+            **{django_trace_header:django_trace_id})
```

Comment:
Those are not equivalent. She wants the _value of_ `django_trace_header`, not the string literal as you have done.